Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -69,10249 +71,4514 @@
-
-        // Left is not less than right
-        Jump isGreaterThan = is32
-            ? m_jit.branchFloat(MacroAssembler::DoubleGreaterThanAndOrdered, left, right)
-            : m_jit.branchDouble(MacroAssembler::DoubleGreaterThanAndOrdered, left, right);
-
-        // NaN
-        if constexpr (is32)
-            m_jit.addFloat(left, right, result);
-        else
-            m_jit.addDouble(left, right, result);
-        Jump afterNaN = m_jit.jump();
-
-        // Left is strictly greater than right
-        isGreaterThan.link(&m_jit);
-        auto isGreaterThanResult = IsMinOrMax == MinOrMax::Max ? left : right;
-        m_jit.moveDouble(isGreaterThanResult, result);
-        Jump afterGreaterThan = m_jit.jump();
-
-        // Left is strictly less than right
-        isLessThan.link(&m_jit);
-        auto isLessThanResult = IsMinOrMax == MinOrMax::Max ? right : left;
-        m_jit.moveDouble(isLessThanResult, result);
-        Jump afterLessThan = m_jit.jump();
-
-        // Left is equal to right
-        isEqual.link(&m_jit);
-        if constexpr (is32 && IsMinOrMax == MinOrMax::Max)
-            m_jit.andFloat(left, right, result);
-        else if constexpr (is32)
-            m_jit.orFloat(left, right, result);
-        else if constexpr (IsMinOrMax == MinOrMax::Max)
-            m_jit.andDouble(left, right, result);
-        else
-            m_jit.orDouble(left, right, result);
-
-        // Continuation
-        afterNaN.link(&m_jit);
-        afterGreaterThan.link(&m_jit);
-        afterLessThan.link(&m_jit);
-#endif
-    }
-
-    PartialResult WARN_UNUSED_RETURN addF32Min(Value lhs, Value rhs, Value& result)
-    {
-        EMIT_BINARY(
-            "F32Min", TypeKind::F32,
-            BLOCK(Value::fromF32(std::min(lhs.asF32(), rhs.asF32()))),
-            BLOCK(
-                emitFloatingPointMinOrMax<float, MinOrMax::Min>(lhsLocation.asFPR(), rhsLocation.asFPR(), resultLocation.asFPR());
-            ),
-            BLOCK(
-                ImmHelpers::immLocation(lhsLocation, rhsLocation) = Location::fromFPR(wasmScratchFPR);
-                emitMoveConst(ImmHelpers::imm(lhs, rhs), Location::fromFPR(wasmScratchFPR));
-                emitFloatingPointMinOrMax<float, MinOrMax::Min>(lhsLocation.asFPR(), rhsLocation.asFPR(), resultLocation.asFPR());
-            )
-        );
-    }
-
-    PartialResult WARN_UNUSED_RETURN addF64Min(Value lhs, Value rhs, Value& result)
-    {
-        EMIT_BINARY(
-            "F64Min", TypeKind::F64,
-            BLOCK(Value::fromF64(std::min(lhs.asF64(), rhs.asF64()))),
-            BLOCK(
-                emitFloatingPointMinOrMax<double, MinOrMax::Min>(lhsLocation.asFPR(), rhsLocation.asFPR(), resultLocation.asFPR());
-            ),
-            BLOCK(
-                ImmHelpers::immLocation(lhsLocation, rhsLocation) = Location::fromFPR(wasmScratchFPR);
-                emitMoveConst(ImmHelpers::imm(lhs, rhs), Location::fromFP
```

Comment:
Is there a reason these aren't also `loadPairPtr`? It looks like it'd boil down to `loadPair32`, and glancing at the layout of the types involved I would imagine it'd work just as well as it does on 64-bit - then we could share the same assertions and remove the `#if`.

Reply:
Looks like the wrong code got selected...this is the line range I'm actually talking about: https://github.com/WebKit/WebKit/pull/19557/files/e48817af0abb6335bd0163cdf680e20127ea7d91#diff-bc737e3614197a55433a8ec56e92e8abcd7bcd2348a099ae1fbdd0d2c9df9e6aR4111-R4114

Reply:
```
 0 calleeSignatureIndex
 4 calleeCode
 8 calleeInstance
16 jsCalleeAnchor
```

Layout seems to be as pictured above, so I guess we can fold into at least one loadPair, but it's not like on 64 bit.